kani-compiler/src/kani_middle/points_to/points_to_analysis.rs

```diff
@@ -202,161 +204,119 @@ impl<'a, 'tcx> Analysis<'tcx> for PointsToAnalysis<'a, 'tcx> {
             match instance.def {
                 // Intrinsics could introduce aliasing edges we care about, so need to handle them.
                 InstanceKind::Intrinsic(def_id) => {
-                    match self.tcx.intrinsic(def_id).unwrap().name.to_string().as_str() {
-                        name if name.starts_with("atomic") => {
-                            match name {
-                                // All `atomic_cxchg` intrinsics take `dst, old, src` as arguments.
-                                // This is equivalent to `destination = *dst; *dst = src`.
-                                name if name.starts_with("atomic_cxchg") => {
-                                    assert_eq!(
-                                        args.len(),
-                                        3,
-                                        "Unexpected number of arguments for `{name}`"
-                                    );
```

Comment:
I'm not familiar with this code, but where are these checks being done now? Shouldn't they be done in `from_str`?

Reply:
@JustusAdam asked a similar question. These checks are implicitly done in the codegen, but I agree that having a centralized way to check those properties is better.

Reply:
Ok, but there are many other checks that aren't done implicitly (or at least they weren't when I was doing the work) like for example the type-checking done in `codegen_intrinsic_simd_extract`. What do you think we should do with them?

Reply:
I am leaning towards the approach of making the the `Intrinsic` enum constructor to accept a function signature (e.g. a list of StableMIR types) that is validated against what we expect it to be